frontend/src/lib/utils.tsx

```diff
@@ -1259,6 +1263,57 @@ export function isPropertyGroup(
     return false
 }
 
+export function isCohortCriteriaGroup(
+    criteria:
+        | CohortCriteriaGroupFilter
+        | CohortCriteriaGroupFilterValue
+        | AnyCohortGroupType[]
+        | undefined
+        | AnyCohortGroupType
+): criteria is CohortCriteriaGroupFilter {
+    if (criteria) {
+        return (
+            (criteria as CohortCriteriaGroupFilter).type !== undefined &&
+            (criteria as CohortCriteriaGroupFilter).values !== undefined
+        )
+    }
+    return false
+}
+
+export function flattenPropertyGroup(
+    flattenedProperties: AnyPropertyFilter[],
+    propertyGroup: PropertyGroupFilter | PropertyGroupFilterValue | AnyPropertyFilter
+): AnyPropertyFilter[] {
+    const obj: AnyPropertyFilter = {}
+    Object.keys(propertyGroup).forEach(function (k) {
+        obj[k] = propertyGroup[k]
+    })
+    if (isValidPropertyFilter(obj)) {
+        flattenedProperties.push(obj)
+    }
```

Comment:
It seems dangerous to mutate the passed in array without documenting it. More so because the return type of `flattenPropertyGroup` implies it creates a new object.

Consider changing the return type to `void` and being explicit about the mutations.

I realize the intent is probably to create beautiful chains eg, `flattenPropertyGroup(..).doX().doY();` but it's general practice for chains to create a new immutable object after each call (eg think `map`, `filter`, etc). Unless it's very explicitly the builder pattern.